packages/sdk/src/lend/providers/morpho/sdk.ts

```diff
@@ -137,27 +91,36 @@ export function calculateBaseApy(vault: any): number {
   }
 }
 
+/**
+ * Parameters for getVault function
+ */
+interface GetVaultParams {
+  /** Market identifier (address and chainId) */
+  marketId: LendMarketId
+  /** Chain manager instance */
+  chainManager: ChainManager
+  /** Optional market allowlist */
+  marketAllowlist?: LendMarketConfig[]
+}
+
 /**
  * Get detailed vault information with enhanced rewards data
- * @param vaultAddress - Vault address
- * @param publicClient - Viem public client
+ * @param params - Named parameters object
  * @returns Promise resolving to detailed vault information
  */
-export async function getVaultInfo(
-  vaultAddress: Address,
-  chainManager: ChainManager,
-): Promise<LendMarket> {
-  try {
-    // 1. Find vault configuration for validation
-    const config = SUPPORTED_VAULTS.find((c) => c.address === vaultAddress)
+export async function getVault(params: GetVaultParams): Promise<LendMarket> {
```

Comment:
Going to keep `vault` here since this function lives within `/morpho/sdk.ts` and wraps the morpho vault functions.